cvat/apps/engine/views.py

```diff
@@ -3156,3 +3182,192 @@ def _import_project_dataset(request, rq_id_template, rq_func, db_obj, format_nam
+            if job:
+                break
+
+        return job
+
+    def _handle_redis_exceptions(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwarggs):
+            try:
+                return func(*args, **kwarggs)
+            except RedisConnectionError as ex:
+                msg = 'Redis service is not available'
+                slogger.glob.exception(f'{msg}: {str(ex)}')
+                return Response(msg, status=status.HTTP_503_SERVICE_UNAVAILABLE)
+        return wrapper
+
+    @method_decorator(never_cache)
+    @_handle_redis_exceptions
+    def retrieve(self, request, pk: str):
+        job = self._get_rq_job_by_id(pk)
+
+        if not job:
+            return HttpResponseNotFound(f"There is no RQ job with specified rq_id: {pk}")
+
+        self.check_object_permissions(request, job)
+
+        serializer = self.get_serializer(job, context={'request': request})
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+    @method_decorator(never_cache)
+    @_handle_redis_exceptions
+    def list(self, request):
+        user_id = request.user.id
+        task_id = request.query_params.get('task_id')
+        action = request.query_params.get('action')
+
+        if task_id and action:
+            assert action == 'create'
+            rq_id = RQIdManager.build(action, 'task', task_id)
+            rq_job = self._get_rq_job_by_id(rq_id)
+
+            filtered_jobs = [] if (not rq_job or rq_job and not self._is_job_owner(rq_job, user_id)) else [rq_job]
+        else:
+            filtered_jobs = self._get_rq_jobs(user_id)
+
+        page = self.paginate_queryset(filtered_jobs)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True, context={'request': request})
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(filtered_jobs, many=True, context={'request': request})
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+    @extend_schema(
+        summary='Cancel RQ job',
+        request=None,
+        responses={
+            '200': OpenApiResponse(description='RQ job has been cancelled'),
+        },
+    )
+    @method_decorator(never_cache)
+    @action(detail=True, methods=['POST'], url_path='cancel')
+    @_handle_redis_exceptions
+    def cancel(self, request, pk: str):
+        rq_job = self._get_rq_job_by_id(pk)
+
+        if not rq_job:
+            return HttpResponseNotFound(f"There is no RQ job with specified rq_id: {pk!r}")
+
+        self.check_object_permissions(request, rq_job)
+
+        if not rq_job.is_started:
+            return HttpResponseBadRequest(f"Only started RQ jobs can be cancelled")
+
+        rq_job.cancel(enqueue_dependents=settings.ONE_RUNNING_JOB_IN_QUEUE_PER_USER)
+
+        return Response(status=status.HTTP_200_OK)
+
+    @method_decorator(never_cache)
+    @_handle_redis_exceptions
+    def destroy(self, request, pk: str):
```

Comment:
I've added this endpoint just because without it user cannot "directly" delete finished/failed jobs. That means that after some time the functionality for managing requests can become useless because it will require iterating over a huge amount of jobs and finding the required one.

Reply:
Well, typically, we use filters and sorting for this. Which are suggested in another comment.